src/main/java/core/basesyntax/db/FruitShopStorage.java

```diff
@@ -0,0 +1,35 @@
+package core.basesyntax.db;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.OperationType;
+import java.util.EnumMap;
+import java.util.Map;
+import java.util.Set;
+
+public class FruitShopStorage implements ShopStorage<Fruit> {
+    private static final String STORAGE_VALUE_ERROR = "Storage value cannot be negative";
+    private final Map<Fruit, Integer> fruitStorage = new EnumMap<>(Fruit.class);
+
+    @Override
+    public int getAmount(Fruit fruit) {
+        return fruitStorage.get(fruit);
+    }
+
+    @Override
+    public void save(Fruit fruit, int value, OperationType operationType) {
```

Comment:
The `save` method in the `ShopStorage` interface should use the generic type `T` instead of `Fruit` to ensure the interface can handle any type `T`. Update the method signature to `void save(T t, int value, OperationType operationType);`.